src/Symfony/Component/Utf8/Tests/AbstractAsciiTestCase.php

```diff
@@ -0,0 +1,830 @@
+     */
+    public function testGetIteratorWithInvalidLimit(int $limit)
+    {
+        static::createFromString('foo|bar|baz')->getIterator($limit)->valid();
+    }
+
+    public function provideInvalidIteratorLimit()
+    {
+        return array(
+            array(-2),
+            array(-1),
+            array(0),
+        );
+    }
+
+    /**
+     * @dataProvider provideGetIteratorData
+     */
+    public function testGetIterator(string $string, array $chunks, int $limit)
+    {
+        $this->assertEquals(
+            $chunks,
+            iterator_to_array(static::createFromString($string)->getIterator($limit))
+        );
+    }
+
+    public function provideGetIteratorData()
+    {
+        return array(
+            array(
+                '',
+                array(),
+                1,
+            ),
+            array(
+                'hello',
+                array(
+                    static::createFromString('h'),
+                    static::createFromString('e'),
+                    static::createFromString('l'),
+                    static::createFromString('l'),
+                    static::createFromString('o'),
+                ),
+                1,
+            ),
+            array(
+                'hello you!',
+                array(
+                    static::createFromString('h'),
+                    static::createFromString('e'),
+                    static::createFromString('l'),
+                    static::createFromString('l'),
+                    static::createFromString('o'),
+                    static::createFromString(' '),
+                    static::createFromString('y'),
+                    static::createFromString('o'),
+                    static::createFromString('u'),
+                    static::createFromString('!'),
+                ),
+                1,
+            ),
+            array(
+                'hell',
+                array(
+                    static::createFromString('h'),
+                    static::createFromString('e'),
+                    static::createFromString('l'),
+                    static::createFromString('l'),
+                ),
+                1,
+            ),
+            array(
+                'hell',
+                array(
+                    static::createFromString('he'),
+                    static::createFromString('ll'),
+                ),
+                2,
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider provideLowercaseData
+     */
+    public function testToLowerCase(string $expected, string $origin)
+    {
+        $instance = static::createFromString($origin)->toLowerCase();
+
+        $this->assertNotSame(static::createFromString($origin), $instance);
+        $this->assertEquals(static::createFromString($expected), $instance);
+        $this->assertSame($expected, (string) $instance);
+    }
+
+    public function provideLowercaseData()
```

Comment:
I would use static methods for all data providers. It avoids PHPUnit to instantiate the TestCase at the beginning of the testsuite to run providers (and it makes it clear that we cannot share state with tests in private properties, because of that).